src/uv-common.c

```diff
@@ -225,6 +225,35 @@ int uv_ip6_name(const struct sockaddr_in6* src, char* dst, size_t size) {
   return uv_inet_ntop(AF_INET6, &src->sin6_addr, dst, size);
 }
 
+int uv_if_indextoname(unsigned int ifindex, char* ifname) {
+#ifdef _WIN32
+  NET_LUID InterfaceLuid;
+  int error = ConvertInterfaceIndexToLuid(ifindex, &InterfaceLuid);
+  if (NO_ERROR != error) {
+    return -error;
+  }
+  char InterfaceName[NDIS_IF_MAX_STRING_SIZE + 1];
+  int status = ConvertInterfaceLuidToAName(InterfaceLuid, InterfaceName, sizeof(InterfaceName));
+  if (NETIO_ERROR_SUCCESS != status) {
+    return -status;
+  }
+  if (strlen(InterfaceName) >= UV_IF_NAMESIZE) {
+    return UV__ENAMETOOLONG;
+  }
+  strncpy(ifname, InterfaceName, UV_IF_NAMESIZE);
+  return 0;
+#else
+  const int saved_errno = errno;
+  errno = 0;
+  if (NULL == if_indextoname(ifindex, ifname)) {
+    assert(errno);
```

Comment:
I agree that if the function returns `NULL`, `errno` should be non-zero, but again tried to be defensive.